### ISSUE TITLE
- Testing

### ISSUE THREAD

**Issue by Dekablade01** [amnmblchllng/survey-list-app#9]
- With the MVC design pattern, most of the logics are in ViewController. 
    - But View Controller is not tested.
- In testing, the response should be mocked. Unit tests should not rely on server-side.

**Comment by amnmblchllng** [comment]
Hi,
- What kind of tests for the ViewController? As it's mostly UI I can't think of Unit Tests for it right now. UI Tests then?
- I agree that mocking is a good idea. Although unit tests can also use server responses (almost integration tests). It would make sense to provide both mock+server I think.

**Comment by Dekablade01** [comment]
> * What kind of tests for the ViewController? As it's mostly UI I can't think of Unit Tests for it right now. UI Tests then?

Yes you're right, what you can assert in UI Tests are `is buttons, images really exists`, `is it navigate to next ViewControler when button did tap`

But what I mean is view controller doesn't contains only UI implementation, but also contains fetching logics eg like

- how can you be sure that the pagegination is not hard coded value like page is always equal to 1 ?
- how can you be sure that the view controller is going to fetch the surveys ?

> * I agree that mocking is a good idea. Although unit tests can also use server responses (almost integration tests). It would make sense to provide both mock+server I think.

- the challenge requires unit tests not Integration tests. 
- Unit test is for developers to test their code is it working as expected or not, while integration is for ops to test that the whole service is working as expected or not.

**Comment by amnmblchllng** [comment]
> - the challenge requires unit tests not Integration tests.
> - Unit test is for developers to test their code is it working as expected or not, while integration is for ops to test that the whole service is working as expected or not.

Yes I know, though tests that use the API can still be unit tests
